### ISSUE TITLE
Sway crashes/freezing when resizing a window

### ISSUE THREAD

**Issue by ZOulhadj** [swaywm/sway#5883]
**Sway Version**
1.5.1

**Sway Log**
https://gist.github.com/ZOulhadj/f52491f74bfbca2ec2547b40e10ff50e

**Sway Stack Trace**
https://gist.github.com/ZOulhadj/aa37b3c0e71f93f53e9f93684c82b333

**Configuration File**
https://gist.github.com/ZOulhadj/eb6893b8217b627c56a43eb5d9ad8c82

**Description**

Whenever resizing a window (either docked or floating) using the mouse or resize mode, sway starts to lag. If I continue to resize the window while the lag occurs, a few seconds later sway freezes entirely. While Sway is frozen, I cannot move my mouse or interact with the system in any way. Audio does, however, continue to play. Around 1-2 minutes after Sway has frozen, I get automatically logged out of my system and brought back to the TTY login screen.

I've noticed that the freezing occurs for applications built to work natively with Wayland as well as those running through XWayland.

From looking at the Sway log, I can see that there does indeed seem to be an error occurring at the very bottom so I wonder if this is related?

I run at a resolution of 5120x1440@60. I reduced my resolution to see if this issue still occurred which it did, unfortunately.

**Comment by Xyene** [comment]
Your stack trace is missing debug symbols. You can get symbols by compiling from source, reproducing the crash and then running `coredumpctl gdb` and then `bt full`.

Here are some instructions to compile from source: https://github.com/swaywm/sway/wiki/Development-Setup#compiling-as-a-subproject

**Comment by ZOulhadj** [comment]
@Xyene Hi, apologies for that. I have now updated the Gist which should now contain the debug symbols.

**Comment by ZOulhadj** [comment]
**Update**

Having added the lines

```
--enable-features=UseOzonePlatform
--ozone-platform=wayland
```

to my `.config/chromium-flags.conf` file it seems that Sway no longer crashes and no lag occurs. So this may possibly be a bug related to Xorg-Wayland package or Chromium instead of Sway. I will continue to test this and report back if anything changes.

**Comment by travankor** [comment]
>So this may possibly be a bug related to the Xorg-Wayland package or Chromium instead of Sway.

if a buggy client (like XWayland) crashes a compositor, it's a bug in the compositor

**Comment by Xyene** [comment]
It's hard to tell because you don't have symbols for `libdrm_nouveau`, `libEGL_mesa` and `nouveau_dri`, but the crash appears to be a failing assertion in nouveau ("nouveau: kernel rejected pushbuf: No such device"). nouveau failing asserts internally strikes me as something that should never happen regardless of what Sway sends at it. @emersion, what do you think?

**Comment by emersion** [comment]
Yes, I agree this sounds like a nouveau bug, Libraries shouldn't assert on runtime errors, so this is likely bad nouveau state due to a nouveau bug. Please report a bug: https://gitlab.freedesktop.org/drm/nouveau/

**Comment by dsko1234** [comment]
I (#5889) have been able to reproduce by resizing xwayland windows.
It never crashed on the first resize, it took significant time of me just sitting there resizing an xwayland window.
I do have logs of this, but it's about 60K Lines of resize transactions, with no info on the actual crash.

As in #5889 described sway doesn't instantly crash for me but rather freezes and just sits there.
Sometimes i have [nouveau.txt](https://github.com/swaywm/sway/files/5721333/nouveau.txt) in my sway log, sometimes only in the log of the programm that i was running.
Its like 350 lines of what i presume to be register/memory states.


I have not yet been able to produce a crash with no involvement of xwayland.

**Comment by Xyene** [comment]
Thanks for testing. Given the identical error message, this does seem like the same nouveau bug as #5889. It'd probably be particularly useful for the nouveau devs if you (or @ZOulhadj) could obtain debug symbols to fill in [the missing stack frames](https://github.com/swaywm/sway/issues/5889#issuecomment-746635532) leading up to the crash, and then opening a ticket at https://gitlab.freedesktop.org/drm/nouveau/

**Comment by cartok** [comment]
Do others also expericence chrome crashes many times a day since some chrome update? Latest sway release was in January so It can not be a change on sway but still this does not happen in other wayland compositors.

**Comment by cartok** [comment]
Chrome update to 105.0.5195.52-1 finally fixed it
